Buildings/Components/Walls/WallTextureBundle.gd

```diff
@@ -0,0 +1,115 @@
+@tool
+extends Resource
+class_name WallTextureBundle
+
+@export var albedo_texture: Texture
+@export var normal_texture: Texture
+
+@export var roughness_texture: Texture : 
+	set(texture):
+		roughness_texture = texture
+		_on_bundled_texture_changed(roughness_texture, metallic_texture, emission_texture)
+@export var metallic_texture: Texture: 
+	set(texture):
+		metallic_texture = texture
+		_on_bundled_texture_changed(roughness_texture, metallic_texture, emission_texture)
+@export var emission_texture: Texture: 
+	set(texture):
+		emission_texture = texture
+		window_shading = true
+		_on_bundled_texture_changed(roughness_texture, metallic_texture, emission_texture)
+
+var window_shading := false
+
+var bundled_texture
+
+
+func _on_bundled_texture_changed(roughness: Texture, metallic: Texture, emission: Texture):
+	# If no textures are set there is no sense in bundling
+	if roughness == null and metallic == null and emission == null: return
```

Comment:
I think there's an edge case here where, if we want to remove all textures, the bundled_texture remains set because we return early here. Might be safer to add `bundled_texture = null` before returning